packages/workbox-cli/src/lib/questions/ask-extensions-to-cache.js

```diff
@@ -0,0 +1,81 @@
+/**
+ * Copyright 2017 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ **/
+
+const assert = require('assert');
+const inquirer = require('inquirer');
+const nodeDir = require('node-dir');
+const path = require('path');
+
+const errors = require('../errors');
+const {ignoredDirectories} = require('../constants');
+
+// The key used for the question/answer.
+const name = 'globPatterns';
+
+/**
+ * @param {string} globDirectory The directory used for the root of globbing.
+ * @return {Promise<[string]>} The unique file extensions corresponding to all
+ * of the files under globDirectory.
+ */
+async function getAllFileExtensions(globDirectory) {
+  const substringsToIgnore = ignoredDirectories.map(
```

Comment:
TODO: Put in a spinner since this might take some time for directories with lots and lots of files.

Reply:
I've added it in https://github.com/GoogleChrome/workbox/pull/965/commits/469ca6f1aead8d1f1b1852e6fe87ded444e34642